admin/services/school-impersonation.service.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+const R = require('ramda')
+
+const schoolDataService = require('../services/data-access/school.data.service')
+const schoolImpersonationValidator = require('../lib/validator/school-impersonation-validator')
+
+const schoolImpersonationService = {}
+
+/**
+ * Validate and assign school to helpdesk session
+ * @param {object} user
+ * @param {string} dfeNumber
+ * @returns {Object}
+ */
+schoolImpersonationService.setSchoolImpersonation = async (user, dfeNumber) => {
+  let validationError
+  validationError = schoolImpersonationValidator.isDfeNumberEmpty(dfeNumber)
+  // returns a validation error if dfeNumber provided is empty
+  if (validationError.hasError()) {
+    return validationError
+  }
+  let school
+  try {
+    school = await schoolDataService.sqlFindOneByDfeNumber(dfeNumber)
+  } catch (ignore) {}
+  // returns a validation error if the school record is not valid
+  validationError = schoolImpersonationValidator.isSchoolRecordValid(school)
```

Comment:
This is another variable, e.g. schoolValidationError.  Should be const.

Reply:
@jon-shipley updated